common/djangoapps/student/views.py

```diff
@@ -1464,13 +1473,38 @@ def create_account_with_params(request, params):
         extended_profile_fields=extended_profile_fields,
         enforce_username_neq_password=True,
         enforce_password_policy=enforce_password_policy,
-        tos_required=tos_required
+        tos_required=tos_required,
     )
 
     with transaction.commit_on_success():
-        ret = _do_create_account(form)
-
-    (user, profile, registration) = ret
+        # first, create the account
+        (user, profile, registration) = _do_create_account(form)
+
+        # next, link the account with social auth, if provided
+        if should_link_with_social_auth:
+            request.social_strategy = social_utils.load_strategy(backend=params['provider'], request=request)
+            social_access_token = params.get('access_token')
+            if not social_access_token:
+                raise ValidationError({
+                    'access_token': [
+                        _("An access_token is required when passing value ({}) for provider.").format(
+                            params['provider']
+                        )
+                    ]
+                })
+            request.session[pipeline.AUTH_ENTRY_KEY] = pipeline.AUTH_ENTRY_REGISTER_API
+            pipeline_user = None
+            error_message = ""
+            try:
+                pipeline_user = request.social_strategy.backend.do_auth(social_access_token, user=user)
+            except HTTPError:
+                error_message = _("The provided access_token is not valid.")
+            except AuthException as auth_exception:
```

Comment:
It's better behavior to catch these in the event that they do occur rather than sending an unfriendly response to the client, which is what I saw when testing last night.  Since the 3rd party authentication is outside of our control, I'd like to be defensive against this.